tests/tpch/test_dask.py

```diff
@@ -306,3 +306,49 @@ def test_query_7(client, dataset_path, fs):
         by=["supp_nation", "cust_nation", "l_year"],
         ascending=True,
     ).compute()
+
+
+@pytest.mark.shuffle_p2p
+def test_query_14(client, dataset_path, fs):
+    """
+    select
+        round(100.00 * sum(case
+            when p_type like 'PROMO%'
+                then l_extendedprice * (1 - l_discount)
+            else 0
+        end) / sum(l_extendedprice * (1 - l_discount)), 2) as promo_revenue
+    from
+        lineitem,
+        part
+    where
+        l_partkey = p_partkey
+        and l_shipdate >= date '1995-09-01'
+        and l_shipdate < date '1995-09-01' + interval '1' month
+    """
+    lineitem = dd.read_parquet(dataset_path + "lineitem", filesystem=fs)
+    part = dd.read_parquet(dataset_path + "part", filesystem=fs)
+
+    shipdate_from = datetime.strptime("1995-09-01", "%Y-%m-%d")
+    shipdate_to = shipdate_from + timedelta(days=365 / 12)
+
+    table = lineitem.merge(part, left_on="l_partkey", right_on="p_partkey", how="inner")
+    table = table[
+        (table.l_shipdate >= shipdate_from) & (table.l_shipdate < shipdate_to)
+    ]
+
+    # Promo revenue by line; CASE clause
+    table["promo_revenue"] = 0.00
+    mask = table.p_type.str.match("PROMO*")
+    table["promo_revenue"] = table.promo_revenue.where(
+        mask, table.l_extendedprice * (1 - table.l_discount)
+    )
+
+    # aggregate promo revenue calculation
+    _ = round(
```

Comment:
nit: Can we move the `round` into the computation?

Reply:
the `dd.Scalar<...>` has doesn't have .round method, but could do with using to_delayed the computing. :+1: